Added rev_history topic template, and fixed replaceable markup in gui…

CowboysFan opened the pull request:
This PR adds the rev_history template topic, and also fixes markup with replaceables in command syntax.

@adellape @bfallonf @tpoitras @tnguyen-rh PTAL and let me know if anything needs to be changed before I merge.

tnguyen-rh commented:
It looks like the build failed due to lacking rev_history directory.
Am i missing something?

adellape commented on _build_cfg.yml:
```diff
@@ -118,7 +118,9 @@ Topics:
         File: scm
-
+  - Name: Revision History
+    File: rev_history
```
I'm guessing these are vestigial from an earlier local test run, causing the current build failure.

adellape commented:
@CowboysFan While you're in the file again, could you comb through the "Code blocks" and "Inline Code or Commands" sections to make sure they jive with the latest (specifically related to replaceables). I think there's some remaining language that may be out of date.

Also, this anchor link in "Inline Code or Commands" doesn't work, at least in GH:

```
See <<Command syntax and examples>> for information on how to show commands and command syntax.
```

CowboysFan commented:
@adellape Thanks, I'll take a look. And yes, the build's broken because I was testing some local builds and forgot to revert the config file back. I'll fix that with my next pass through.

CowboysFan commented:
@adellape @tnguyen-rh @tpoitras @bfallonf Okay, I've made some updates and reverted the build file back. It's building locally, but Travis CI seems to be stuck. Let me know if anything else needs to be changed before I merge it.

adellape commented on contributing_to_docs/doc_guidelines.adoc:
```diff
@@ -215,15 +215,7 @@ For all of the system blocks including table delimiters, use four characters. Fo
-* Do NOT use any markup in code blocks; code blocks generally do not accept any markup
-
-* Try to use callouts to provide information on what the output represents when required
+Code blocks are used to command usage, syntax, and examples of command screen outputs or configuration files. 
```
s/used to/used to show/ ?

adellape commented on contributing_to_docs/doc_guidelines.adoc:
```diff
@@ -248,21 +240,20 @@ $ lorem.sh
-=== Inline Code or Commands
-Do NOT show full commands or command syntax inline within a sentence. See <<Command syntax and examples>> for information on how to show commands and command syntax.
+=== Command syntax and examples
+Code blocks are used to show command usage, syntax, and examples of command screen outputs or configuration files. Code blocks can be used in a few different ways:
```
It seems weird to start both of these back to back sections with the same sentence (line 218 for "Code blocks").

adellape commented on contributing_to_docs/doc_guidelines.adoc:
```diff
@@ -248,21 +240,20 @@ $ lorem.sh
-The main distinction between showing command syntax and example is that a command syntax should just show customers how to use the command without real values. An example on the other hand should show the command with actual values with an example output of that command, where applicable.
+* If the command or command syntax warrants a full example, wrap it inside an example block, and use actual example values. Do NOT use <replaceables> in command example outputs.
+
+* Try to use callouts to provide information on what the output represents when required
```
We mention callouts here and in another spot below, yet we don't explain anywhere what they are or show how to use them.

bfallonf replied:
@CowboysFan I'd suggest this "Command Syntax and Examples" section above be condensed a little. We've got two lists (one using numbers, and the other dot points) that are for the same purpose. The list items are even quite similar. I'd suggest that the first (numbered) list is redundant, as the info there is either in the little intro para, or the dot point list.

tnguyen-rh commented on contributing_to_docs/templates/topic_template.adoc:
```diff
@@ -12,4 +12,4 @@ toc::[]
 --All topics should include an Overview section that introduces the topic. Remove this and add a brief introduction here.--
 
-== Section Title
+== Section Title
```
Probably removing the newline from eof was a typo?
It seems asciidoctor doesn't care, but for future-proofing against lame tools, we probably want to maintain eof newline.
Reverting this hunk also makes for a smaller diff, always a good thing.  :)

bfallonf commented:
@CowboysFan remember to fix the "Quick markup reference" table so that it doesn't talk about sidebar blocks in the replaceables section.

adellape commented:
@CowboysFan Some outstanding comments here.

adellape commented:
@CowboysFan This needs a rebase now.

CowboysFan commented:
@adellape Yeah, I realize that. Got some other higher priority items on the list, but I'll definitely rebase when I fix all this up.

CowboysFan commented:
@adellape @bfallonf @tnguyen-rh Sorry, I got sidetracked with other things. I'll try to get all this fixed up and update this PR. Apologies for the delay.

CowboysFan commented:
@adellape @bfallonf @tnguyen-rh @tpoitras Please take a look and see if I've covered everything from these comments. I hope I addressed all the issues, but let me know I still missed something.

bfallonf commented:
@CowboysFan This looks good to me.

adellape commented on contributing_to_docs/doc_guidelines.adoc:
```diff
@@ -421,24 +434,22 @@ The `GET` operation can be used to do something.
+a|Use the *`IP_ADDRESS*` environment variable for the server IP address.
 
-The `*MAX_PODS*` parameter limits the number of pods you can have.
+The *`MAX_PODS`* parameter limits the number of pods you can have.
```
These are now backwards. It was correct the way it was (and how we've been using it). From http://asciidoctor.org/docs/user-manual/#mono:

"Monospaced text can be bold and italicized, as long as the markup sets are entered in the right order. The monospace markup must be the outermost set, then the bold set, and the italic markup must always be the innermost set."

CowboysFan commented:
@adellape True, but I think this way it works both in GitHub and with the build mechanism. This was an issue brought up where the rendering is not correct in GitHub. I think this way we won't have any issues, but let me know if you guys think it breaks something.

adellape commented:
@CowboysFan So, I did verify that the recommended syntax...

```
`*MAX_PODS*`
```

and the proposed change syntax...

```
*`MAX_PODS`*
```
...do both seem to render correctly (i.e., bold and monospaced) in local builds. Granted, this was just after some fairly simple testing, could definitely be some edge cases I'm not thinking of. And in GitHub's rendered adoc previews, the recommended syntax appears as monospaced with visible asterisks around it, while the proposed syntax appears as bold monospace. 

I'm still hesitant about this change, though. Comparing the pros/cons/unknowns of the proposed change:

Pros:
- GitHub rendered adoc previews look slightly nicer. This seems a bit moot though, considering we can't rely on GH's previews anyway since we use ifdefs. E.g., a heavily-conditionalized topic such as https://github.com/openshift/openshift-docs/blob/master/admin_guide/install/quick_install.adoc is pretty much useless in GH previews, and that's just one of many examples.

Cons: 
- Inconsistent with standard practices (http://asciidoctor.org/docs/user-manual/#mono as mentioned previously).
- Would need to safely change the 400+ instances of the previous usage and verify the changes.
- Contributors/devs by now seem to have gotten used to the existing style (based on standard practice) and would have to learn/remember to switch usage going forward (and/or we'd have to correct).
- Contributors coming in fresh would have to be made aware of this quirk instead of being able to rely on standard practice.

Unknowns:
- How do Customer Portal builds behave with this change? Has anyone tested it? I would hate to make this switch, only to later find out that it's yet another thing that the Portal builds don't like.
- Potentially impacts the portability of our source, as far as how other future process may behave. Sticking with the recommended syntax at least hedges our bets a bit.

The sticking points for me are that we'd be deviating from the recommended practice seemingly just to make a low-priority (in my opinion) render look slightly nicer. I say low-priority since the first-class builds that we push people towards are the docs.openshift.* sites and the Customer Portal builds, plus the GH previews are unreliable anyway (per above mentioned ifdefs) and I therefore honestly put little stock in them. And an important question we should at the very least answer/verify is how the change looks on the Portal.

cc @tnguyen-rh @tpoitras @bfallonf @vikram-redhat

CowboysFan commented:
@adellape Okay, I'll leave these as-is and not worry about how they render in GitHub. That way we stay with the standard AsciiDoc practices.

adellape commented:
:+1: Sounds good. Thanks.

CowboysFan commented:
@adellape @tpoitras @bfallonf @tnguyen-rh @vikram-redhat Unless there is something else  missing, I think this is ready to be merged. Thanks.

adellape commented:
LGTM
